stripe/src/main/java/com/stripe/android/paymentsheet/ui/BillingAddressView.kt

```diff
@@ -315,6 +315,12 @@ internal class BillingAddressView @JvmOverloads constructor(
         } else {
             PostalCodeConfig.Global
         }
+
+        if (country?.code == "US") {
+            viewBinding.postalCodeLayout.hint = resources.getString(R.string.acc_label_zip_short)
+        } else {
+            viewBinding.postalCodeLayout.hint = resources.getString(R.string.address_label_postal_code)
+        }
```

Comment:
On the surface this looks like non-idiomatic string resource usage (the string resource should represent the same thing just in different languages). I think given the state of our translations (😬) this is fine, but we should probably track this so that we can represent this as a single string resource.

Reply:
In the case where you change the language of the phone, things update as you would expect.  The problem is when you set the selected country.

Are you thinking use something more like this:
```Configuration conf = getResources().getConfiguration();
conf.locale = new Locale("gb");
DisplayMetrics metrics = new DisplayMetrics();
getWindowManager().getDefaultDisplay().getMetrics(metrics);
Resources resources = new Resources(getAssets(), metrics, conf);
String str = resources.getString(id);```

Reply:
I see this is also how it is used in the updateStateView

Reply:
If so I wonder if we should just update the application resources when there is a country change, or if we should be more exact in which resources we change?